hibernate-core/src/main/java/org/hibernate/dialect/H2Dialect.java

```diff
@@ -435,10 +435,8 @@ public boolean supportsTuplesInSubqueries() {
 	}
 	
 	@Override
-	public boolean dropConstraints() {
-		// We don't need to drop constraints before dropping tables, that just leads to error
-		// messages about missing tables when we don't have a schema in the database
-		return false;
+	public boolean supportsIfExistsAfterAlterTable() {
```

Comment:
I had to borrow all of this from HSQLDialect to work correctly
```suggestion
	// Do not drop constraints explicitly, just do this by cascading instead.
	@Override
	public boolean dropConstraints() {
		return false;
	}

	@Override
	public String getCascadeConstraintsString() {
		return " CASCADE ";
	}

	// CASCADE has to be AFTER IF EXISTS in case it's after the tablename
	@Override
	public boolean supportsIfExistsAfterTableName() {
		return false;
	}

	@Override
	public boolean supportsIfExistsBeforeTableName() {
```

Reply:
You will also have to remove the prior `supportsIfExistsAfterTableName` in this file.